index.d.ts

```diff
@@ -782,6 +782,7 @@ declare module 'mongoose' {
 
     /** Adds a discriminator type. */
     discriminator<D extends Document>(name: string, schema: Schema, value?: string): Model<D>;
+    discriminator<T extends Document, U extends Model<T, TQueryHelpers>, TQueryHelpers = {}>(name: string, schema: Schema<T, U>, value?: string): U;
```

Comment:
Is there any need for `TQueryHelpers` here? It doesn't seem necessary since it's just in the `Model` generic.

Reply:
That's true! We don't need `TQueryHelpers` here.

Thanks for careful reading. 😃